### PR TITLE
job-list: handle per-resource "cores" special cases

### PR DESCRIPTION
**Pull request by chu11**
second attempt at this after failed attempt in #4601

### CODE REVIEW

**Comment by chu11** [comment]
re-pushed, fixed a missing && chain and missed HAVE_JQ in a few new tests

File: t/t2260-job-list.t [hunk 1]
```diff
@@ -711,6 +711,22 @@ test_expect_success HAVE_JQ 'flux job list outputs ntasks correctly (tasks-per-n
         echo $obj | jq -e ".ntasks == 6"
 '
 
+test_expect_success 'flux jobs list outputs ntasks correctly (per-resource.type=node)' '
```
Comment (grondo) [inline t/t2260-job-list.t]: `flux job list` not `flux jobs list`?

File: t/t2260-job-list.t [hunk 2]
```diff
@@ -727,6 +730,54 @@ test_expect_success 'flux jobs list outputs ntasks correctly (per-resource.type=
...
+        expected=$((ncores * 2)) &&
+        echo ${expected} > per_resource_type_core_ntasks1.exp &&
+        obj=$(flux job list -s inactive | grep $jobid) &&
+        echo $obj | jq -e ".ntasks == ${expected}"
+'
+
+# N.B. As of this test writing, tasks-per-core uses
+# per-resource.type=core.  But write direct test in case of future
+# changes.
+test_expect_success HAVE_JQ 'flux jobs list outputs ntasks correctly (cores / per-resource.type=core)' '
```
Comment (grondo) [inline t/t2260-job-list.t:767]: `s/flux jobs list/flux job list/`

**Review by grondo — APPROVED** [review]
LGTM. Just a couple potential typos noted inline.

**Comment by chu11** [comment]
thanks @grondo, re-pushed with those minor tweaks, setting MWP

**Pull request merged**